kernel/mem/pmm.c

```
@@ -0,0 +1,99 @@
+
+void pmm_unset_block(uint32_t bit) {
+	memory_bitmap[bit/32] &= ~(1 << (bit % 32));
+}
+
+int pmm_find_free_blocks(uint32_t blocks) {
+	if (blocks == 0) return -1;
+
+	for (uint32_t i = 0; i < max_blocks / 32; i++) {
+		if (memory_bitmap[i] != 0xFFFFFFFF) {
+			for (uint32_t j = 0; j < 32; j++) {
+				int bit = (1 << j);
```

Cppcheck (2.20.0)

[error] Shifting signed 32-bit value by 31 bits is undefined behaviour

(shiftTooManyBitsSigned)